doc/nrf/app_dev/wifi_coex/index.rst

```diff
@@ -1,103 +1,113 @@
+Integration considerations
+**************************
 
-Requirements
-************
+The |NCS| provides the coexistence feature by providing SR protocol drivers capable of calling coexistence API and coexistence implementations implementing the mentioned API.
+The coexistence feature are provided in following ways:
 
-If your application uses SR protocols and requires coexistence with Wi-Fi, you must enable :ref:`nrfxlib:mpsl_cx` in the :ref:`nrfxlib:mpsl` (MPSL) library.
+   * :ref:`ug_radio_coex_mpsl_cx_based`
+   * :ref:`ug_radio_coex_bluetooth_only_based`
 
-Wi-Fi Coexistence and MPSL
-==========================
 
-The MPSL library provides an implementation of the Wi-Fi Coexistence for SR applications.
-To use this implementation, your application must use a protocol driver that enables the Wi-Fi Coexistence (CX) feature.
-The library provides multiprotocol support, but you can also use it in applications that require only one protocol.
+.. _ug_radio_coex_mpsl_cx_based:
 
-Currently, the following protocols use the Wi-Fi Coexistence support provided by MPSL:
+Coexistence based on MPSL CX API
+================================
 
-   * :ref:`ug_thread`
-   * :ref:`ug_zigbee`
+It is based on :ref:`nrfxlib:mpsl_cx` provided by the :ref:`nrfxlib:mpsl` (MPSL) library and is recommended for new designs.
 
-.. _ug_radio_cx_requirements:
+The following SR protocols are compatible with the coexistence based on MPSL CX API:
 
-Enabling MPSL
--------------
+   * :ref:`Bluetooth LE <ug_ble_controller>` if implementation used is :ref:`SoftDevice Controller <nrfxlib:softdevice_controller>`
+   * Protocols based on :ref:`nrfxlib:nrf_802154` such as :ref:`ug_thread` and :ref:`ug_zigbee`.
 
-The MPSL library provides an API to handle signals coming from the PTA and to control requests sent to PTA.
-See the :ref:`nrfxlib:mpsl_lib` in the nrfxlib documentation for details.
+This way of providing coexistence applies to the following implementations:
 
-Enable support for the MPSL implementation in |NCS| by setting the :kconfig:option:`CONFIG_MPSL` Kconfig option to ``y``.
+   * :ref:`ug_radio_mpsl_cx_nrf700x`
+   * :ref:`ug_radio_mpsl_cx_generic_3wire`
+   * :ref:`ug_radio_mpsl_cx_custom`
 
-.. note::
-   When using the nRF5340, add the :kconfig:option:`CONFIG_MPSL` option only for the NET core.
+The |NCS| provides a wrapper that configures Wi-Fi Coexistence based on devicetree (DTS) and Kconfig information.
 
-Enabling Wi-Fi Coexistence
---------------------------
+Common requirements to use coexistence based on MPSL CX API:
 
-To enable Wi-Fi Coexistence, do as follows:
+1. Make sure that Kconfig option :kconfig:option:`CONFIG_MPSL` is enabled.
+   Please note that some protocol drivers like :ref:`SoftDevice Controller <nrfxlib:softdevice_controller>` enable this option by default.
+2. Make sure that Kconfig option :kconfig:option:`CONFIG_MPSL_CX` is enabled.
+3. Make sure you meet requirements specific to selected coexistence implementation.
```

Comment:
This statement is a bit vague in its current shape. What do you think about adding an example to help the reader understand what kind of requirements could be specific to the selected implementation? I mean something like:
> For example in case of nRF700x, additional Kconfig option :kconfig:option:`CONFIG_MPSL_CX_NRF700X` must be enabled. This requirement is specific for nRF700x coexistence implementation. Other coexistence implementations might have different requirements or have no additional requirements at all. For further details on requirements specific to selected coexistence implementation refer to the documentation of that implementation.

Reply:
All these requirements are described for given implementation, please see chapter Enabling nRF700x Wi-Fi coexistence, there is about CONFIG_MPSL_CX_NRF700X.

Reply:
> All these requirements are described for given implementation

Yes, but the original idea of this bullet point "Make sure you meet requirements specific to selected coexistence implementation." was to make the user aware that these generic requirements common for all implementations might not be enough. If you want to avoid duplication then maybe it's enough to add something like:

> Make sure you meet requirements specific to selected coexistence implementation. For details on requirements specific to selected coexistence implementation refer to the documentation of that implementation.

In my opinion without any additional example or information this information doesn't help much.

Reply:
@jciupis changed a bit. Is it enough?

Reply:
Yes, it's very good now 👍